### PR TITLE
enhancement: split out protocol service

### PR DESCRIPTION
**Pull request by axejintao**

### CODE REVIEW

File: src/protocols/sushi/SushiswapService.ts [hunk 1]
```diff
@@ -1,48 +1,77 @@
...
 		const [totalAllocPoint, sushiPerBlock, poolInfo, tokenPrice] = await Promise.all([
 			masterChef.totalAllocPoint() as number,
 			masterChef.sushiPerBlock() as number,
 			masterChef.poolInfo(poolId) as PoolInfo,
 			this.priceService.getTokenPriceData(TOKENS.SUSHI),
 		]);
 		const depositToken = new ethers.Contract(poolInfo.lpToken, erc20Abi, chain.provider);
-		const poolBalance = (await depositToken.balanceOf(contract)) / 1e18;
+		const poolBalance = (await depositToken.balanceOf(pair)) / 1e18;
+		console.log(poolBalance);
```
Comment (ethkaneki) [inline src/protocols/sushi/SushiswapService.ts:59]: 🔪

File: src/cache/CacheService.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+	get<T>(key: string): T | undefined {
+		return this.cache.get(key);
+	}
+
+	set<T>(key: string, value: T): void {
+		this.cache.set(key, value);
+	}
+
+	getCacheKey(...parts: string[]): string {
```
Comment (ethkaneki) [inline src/cache/CacheService.ts:25]: nit: Could maybe move this into a util or make static

**Review by ethkaneki — APPROVED** [review]

File: src/protocols/sushi/SushiswapService.ts (hunk 1 above)
Comment (axejintao) [reply src/protocols/sushi/SushiswapService.ts:59]: he he xd!!!!

**Comment by axejintao** [comment]
nits addressed, subsequent commit was for staging failures on missing packages. merging feel free to leave any comments if last two commits have issues.

**Pull request merged**